openpype/hosts/maya/plugins/create/create_model.py

```diff
@@ -1,26 +1,43 @@
 from openpype.hosts.maya.api import plugin
+from openpype.lib import (
+    BoolDef,
+    TextDef
+)
 
 
-class CreateModel(plugin.Creator):
+class CreateModel(plugin.MayaCreator):
     """Polygonal static geometry"""
 
-    name = "modelMain"
+    identifier = "io.openpype.creators.maya.model"
     label = "Model"
     family = "model"
     icon = "cube"
     defaults = ["Main", "Proxy", "_MD", "_HD", "_LD"]
+
     write_color_sets = False
     write_face_sets = False
-    def __init__(self, *args, **kwargs):
-        super(CreateModel, self).__init__(*args, **kwargs)
-
-        # Vertex colors with the geometry
-        self.data["writeColorSets"] = self.write_color_sets
-        self.data["writeFaceSets"] = self.write_face_sets
 
-        # Include attributes by attribute name or prefix
-        self.data["attr"] = ""
-        self.data["attrPrefix"] = ""
+    def get_instance_attr_defs(self):
 
-        # Whether to include parent hierarchy of nodes in the instance
-        self.data["includeParentHierarchy"] = False
+        return [
+            BoolDef("writeColorSets",
+                    label="Write vertex colors",
+                    tooltip="Write vertex colors with the geometry",
+                    default=self.write_color_sets),
+            BoolDef("writeFaceSets",
+                    label="Write face sets",
+                    tooltip="Write face sets with the geometry",
+                    default=self.write_face_sets),
+            BoolDef("includeParentHierarchy",
+                    label="Include Parent Hierarchy",
+                    tooltip="Whether to include parent hierarchy of nodes in "
+                            "the publish instance",
+                    default=False),
+            TextDef("attr",
+                    label="Custom Attributes",
+                    default="",
+                    placeholder="attr1, attr2"),
+            TextDef("attrPrefix",
+                    label="Custom Attributes Prefix",
+                    placeholder="prefix1, prefix2")
```

Comment:
This is exactly that, or what do you mean? Do you mean to re-label them?
I believe this is a one to one match with the name of the flag for the maya alembic exporter I believe.

Reply:
they are different things: https://github.com/tokejepsen/maya-alembic-export/blob/master/alembic_export.py#L163
>
        userAttrPrefix (list of str, optional): Prefix filter for determining
            which user attributes to write out. Defaults to [].
        userAttr (list of str, optional): Specific user attributes to write
            out. Defaults to [].
        attr (list of str, optional): A specific geometric attribute to write
            out. Defaults to [].
        attrPrefix (list of str, optional): Prefix filter for determining which
            geometric attributes to write out. Defaults to ["ABC_"].

Reply:
🤯 What!? Haha, I had never noticed those were there.

I'm a bit confused too actually. What is the difference? One is a 'geometric attribute' and the other is a 'user attribute'. [It defines where it writes it to in the Alembic file](https://github.com/alembic/alembic/blob/a4b60eadb08500ea8ab7af761ddb3377c5772f30/maya/AbcExport/MayaUtility.h#L235-L241) 

- attr + attrPrefix go into `.geomArbParams`
- userAttr + userAttrPrefix go into `.userProperties`

But then the question becomes what does that mean for the resulting Alembic file. How does it behave differently and which should you choose when?

In [OpenPype we've always used `attr` and `attrPrefix`](https://github.com/ynput/OpenPype/blob/f4d22e6ac671245dab4c4f6831ea6f057ca088c6/openpype/hosts/maya/plugins/publish/extract_pointcache.py#L57-L58) and our studio has done that for years too. For example the `cbId` attribute for nodes is there too which as I read this now makes more sense in `.userProperties` maybe?

It influences for example [how something comes into Katana](https://github.com/ezequielmastrasso/EZLookdevTools/issues/106#issuecomment-544480777). But still, how do I decide which should go where?

And if we want to be that picky. That'd almost make the `includeUserDefinedAttributes` automatic 'add my custom attributes in the export' more complicated since that can't really decide which attribute should go where, right? @tokejepsen what do you think?

Reply:
Yeah I actually also just discovered this difference last year on my prior studio that we had this render procedurals that were reliant on how you exported the attributes as otherwise they weren't readable by the procedural...
ChatGPT does a pretty good job at answering this as well (I didn't even type the question that correct and still understood it perfectly):
![image](https://user-images.githubusercontent.com/4348536/230240400-0703eb9f-0dd6-4a7c-8a21-ad09627c5f4f.png)

Reply:
On any case, it's good to try make the cache exporter interface as thin as possible to the internal API to accommodate for any possible workflows. On my prior studio we exposed most of the Alembic options (as Advanced tabs) because in some cases there was a specific intend to write the attribute as user or compound for specific DCCs or applications expecting it that way

Reply:
Sounds like there are definitely use cases for this. I'll try and make a dedicated issue for this. But I'd like to keep it separate from this PR to keep the functional changes minimal and just closer to a refactor of just the publisher. (It's already a massive PR)

Reply:
Yeah definitely, thank you!!

Reply:
Created an issue here: https://github.com/ynput/OpenPype/issues/4812